frontend/src/components/db/SearchableDbSourceTree/utils.ts

```diff
@@ -81,3 +83,13 @@ export const handleTreeChange = ({
 
   setBreadcrumbs(crumbs);
 };
+
+export function isSearchMatch(searchField: string, searchTerm: string) {
+  return searchField.toLowerCase().includes(searchTerm.toLowerCase());
+}
+
+export function isDbSourceTreeLeafNodeData(
+  node: DbSourceTreeNode | DbSourceTreeLeafNode,
+): node is DbSourceTreeLeafNode {
+  return (node as DbSourceTreeLeafNode).fileName !== undefined;
```

Comment:
this casting looks unsafe. What about the case when this is an instance of `DbSourceTreeNode`? Maybe we can include type guards here.

Reply:
Happy to do another way. But what do you think is unsafe? `fileName` is the specific property that *must* exist on a leaf node and the only property that distinguishes it from the `DbSourceTreeNode` type on which it can be undefined.

Reply:
yes, but TypeScript is saying that there's a possibility that this is not a `DbSourceTreeLeafNode`. If it's `undefined` for example, this would crash the app. Basically casting using the `as` keyword ignores this protection, but we could keep it if we introduce runtime [type guards instead](https://www.typescriptlang.org/docs/handbook/advanced-types.html).